utils/jwt_with_key_rotation.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import * as jwt from "jsonwebtoken";
+import { ulid } from "ulid";
+
+import { pipe } from "fp-ts/lib/function";
+import * as TE from "fp-ts/TaskEither";
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+
+import { Second } from "@pagopa/ts-commons/lib/units";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+
+// --------------------
+
+const alg = "RS256";
+
+// -----------------------
+// Public methods
+// -----------------------
+
+/**
+ * JWT Generation
+ */
+export type GetGenerateJWT = <T extends Record<string, unknown>>(
+  issuer: NonEmptyString,
+  primaryPrivateKey: NonEmptyString
+) => (payload: T, ttl: Second) => TE.TaskEither<Error, NonEmptyString>;
+
+export const getGenerateJWT: GetGenerateJWT = (
+  issuer,
+  primaryPrivateKey
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+) => (payload, ttl) =>
+  pipe(
+    TE.taskify<Error, string>(cb =>
+      jwt.sign(
+        payload,
+        primaryPrivateKey,
+        {
+          algorithm: alg,
+          expiresIn: `${ttl} seconds`,
+          issuer,
+          jwtid: ulid()
+        },
+        cb
+      )
+    )(),
+    TE.mapLeft(E.toError),
+    TE.chain(
+      TE.fromPredicate(NonEmptyString.is, () => E.toError("Token is empty."))
+    )
+  );
+
+/**
+ * JWT Validation
+ */
+export type GetValidateJWT = (
+  issuer: NonEmptyString,
+  primaryPublicKey: NonEmptyString,
+  secondaryPublicKey?: NonEmptyString
+) => (token: NonEmptyString) => TE.TaskEither<Error, jwt.JwtPayload>;
+
+export const getValidateJWT: GetValidateJWT = (
+  issuer,
+  primaryPublicKey,
+  secondaryPublicKey
+) => (token): TE.TaskEither<Error, jwt.JwtPayload> =>
+  pipe(
+    token,
+    validateJWTWithKey(issuer, primaryPublicKey),
+    TE.orElse(err =>
```

Comment:
Could make sense to proceed only if the error is a JsonWebTokenError - invalid signature ?

Reply:
I agree with @fabriziopapi

Reply:
Right. Done in [[#IOCIT-301] second validation when error is invalid signature](https://github.com/pagopa/io-functions-lollipop/pull/14/commits/6105c1d9a7905b5235e4941665fcff499fc8ac77)